dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/OracleSqlList.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.oracle;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.edwmigration.dumper.application.dumper.connector.oracle.QueryGroup.StatsSource.AWR;
+import static com.google.edwmigration.dumper.application.dumper.connector.oracle.QueryGroup.StatsSource.STATSPACK;
+import static com.google.edwmigration.dumper.application.dumper.connector.oracle.QueryGroup.TenantSetup.MULTI_TENANT;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.edwmigration.dumper.application.dumper.connector.oracle.task.StatsJdbcTask;
+import com.google.edwmigration.dumper.application.dumper.task.Task;
+import java.time.Duration;
+import java.util.List;
+import java.util.function.Function;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+/** Represents a list of oracle-stats SQL files that share a common QueryGroup. */
+@AutoValue
+@ParametersAreNonnullByDefault
+abstract class OracleSqlList {
```

Comment:
Optional: I would suggest avoiding suffixing the class `List`, if it does not implement the `List` interface. Such a naming is a bit surprising.

Reply:
Done, I changed the suffix from `List` to `Listing`